src/visitors.rs

```diff
@@ -83,18 +83,22 @@ impl<'a, 'b: 'a, 'tcx: 'b> MirVisitor<'a, 'b, 'tcx> {
     /// Use the local and global environments to resolve Path to an abstract value.
     /// For now, statics and promoted constants just return Top.
     /// If a local value cannot be found the result is Bottom.
-    fn lookup_path(&mut self, path: Path) -> AbstractValue {
-        // todo: rather than simply clone the value, also specialize it with the current path condition
-        let local_val = self.current_environment.value_at(&path).clone();
-        if local_val.is_bottom() {
+    fn lookup_path_and_refine_result(&mut self, path: Path) -> AbstractValue {
+        let refined_val;
+        {
+            let bottom = abstract_value::BOTTOM;
```

Comment:
Was this scope introduced for the borrow checker? I guess a case where NLL didn't help?

Also, matter of personal preference, but I think you can do:

```
  let refined_val = {
       ...
       local_val.refine_with(&self.path_conditions, self.current_span)
   }
```

Reply:
Any unnatural code mangling you see is usually forced upon me by the borrow checker.